lnbits/extensions/boltz/boltz.py

```diff
@@ -0,0 +1,458 @@
+            status_code=HTTPStatus.INTERNAL_SERVER_ERROR, detail="vout_cnt not found"
+        )
+    return tx, txid, vout_cnt, vout_amount
+
+
+async def send_onchain_tx(tx: Transaction):
+    res = httpx.post(
+        MEMPOOL_SPACE_URL + "/api/tx",
+        headers={"Content-Type": "text/plain"},
+        data=hexlify(tx.serialize()),
+        timeout=40,
+    )
+    handle_request_errors(res)
+    return res
+
+
+# send on on-chain, receive lightning
+async def create_swap(data: CreateSubmarineSwap) -> SubmarineSwap:
+    swap_id = urlsafe_short_hash()
+    try:
+        payment_hash, payment_request = await create_invoice(
+            wallet_id=data.wallet,
+            amount=data.amount,
+            memo=f"submarine swap of {data.amount} sats on boltz.exchange",
+            extra={"tag": "boltz", "swap_id": swap_id},
+        )
+    except Exception as e:
+        raise HTTPException(status_code=HTTPStatus.INTERNAL_SERVER_ERROR, detail=str(e))
+
+    refund_privkey = ec.PrivateKey(os.urandom(32), True, net)
+    refund_pubkey_hex = hexlify(refund_privkey.sec()).decode("UTF-8")
+
+    res = create_post_request(
+        BOLTZ_URL + "/createswap",
+        {
+            "type": "submarine",
+            "pairId": "BTC/BTC",
+            "orderSide": "sell",
+            "refundPublicKey": refund_pubkey_hex,
+            "invoice": payment_request,
+        },
+    )
+    logger.info(f" - created swap, boltz_id: {res['id']}. wallet: {data.wallet}")
+    return SubmarineSwap(
+        id=swap_id,
+        time=getTimestamp(),
+        wallet=data.wallet,
+        amount=data.amount,
+        refund_privkey=refund_privkey.wif(net),
+        refund_address=data.refund_address,
+        boltz_id=res["id"],
+        status="pending",
+        address=res["address"],
+        expected_amount=res["expectedAmount"],
+        timeout_block_height=res["timeoutBlockHeight"],
+        bip21=res["bip21"],
+        redeem_script=res["redeemScript"],
+    )
+
+
+def get_fee_estimation() -> int:
+    # hardcoded maximum tx size, in the future we try to get the size of the tx via embit (not possible yet)
+    tx_size_vbyte = 200
+    mempool_fees = get_mempool_fees()
+    return mempool_fees * tx_size_vbyte
+
+
+# claim tx for reverse swaps
+# refund tx for normal swaps
+async def create_onchain_tx(
+    swap: Union[ReverseSubmarineSwap, SubmarineSwap], mempool_lockup_tx
+) -> Transaction:
+
+    if type(swap) == SubmarineSwap:
+        current_block_height = get_mempool_blockheight()
+        if current_block_height <= swap.timeout_block_height:
+            msg = f"refund not possible, timeout_block_height ({swap.timeout_block_height}) is not yet exceeded ({current_block_height})"
+            raise HTTPException(status_code=HTTPStatus.METHOD_NOT_ALLOWED, detail=msg)
+        privkey = ec.PrivateKey.from_wif(swap.refund_privkey)
+        preimage = b""
+        onchain_address = swap.refund_address
+        sequence = 0xFFFFFFFE
+    else:
```

Comment:
What is this case? Please add a comment to the code to explain.

Reply:
added an explanation bfdbdfe1